client/lib/html-toolbar/index.js

```diff
@@ -0,0 +1,228 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { Env } from 'tinymce/tinymce';
+import { noop, reduce } from 'lodash';
+
+/**
+ * Return an object containing the textarea content split into:
+ * - `before` selection
+ * - (`inner`) selection
+ * - `after` selection
+ *
+ * @param {Element} textarea The editor textarea.
+ * @returns {Object} The split textarea content.
+ */
+export const splitSelectedContent = textarea => {
+	const { selectionEnd, selectionStart, value } = textarea;
+	return {
+		before: value.substring( 0, selectionStart ),
+		inner: value.substring( selectionStart, selectionEnd ),
+		after: value.substring( selectionEnd, value.length ),
+	};
+};
+
+/**
+ * Set the cursor position after the insertion of new content into a textarea.
+ *
+ * @param {Element} textarea The editor textarea.
+ * @param {Number} selectionEnd The textarea.selectionEnd value BEFORE the content insertion.
+ * @param {Number} insertedContentLength The length of the inserted content.
+ */
+export const setCursorPosition = ( textarea, selectionEnd, insertedContentLength ) => {
+	textarea.selectionEnd = textarea.selectionStart = selectionEnd + insertedContentLength;
+};
+
+/**
+ * Insert content into a textarea.
+ *
+ * @param {Element} textarea The target textarea.
+ * @param {String} before The content before the selection.
+ * @param {String} content The new content to insert.
+ * @param {String} after The content after the selection.
+ * @param {Function} onInsert A function to call after the insertion.
+ */
+export const insertContent = ( textarea, before, content, after, onInsert = noop ) => {
+	const { selectionEnd, value } = textarea;
+	const newContent = before + content + after;
+
+	if ( Env.gecko ) {
+		// In Firefox, execCommand( 'insertText' ), needed to preserve the undo stack,
+		// always moves the cursor to the end of the content.
+		// A workaround involving manually setting the cursor position and inserting the editor content
+		// is needed to put the cursor back to the correct position.
+		textarea.value = newContent;
+		textarea.focus();
+		document.execCommand( 'insertText', false, content );
+		setCursorPosition( textarea, selectionEnd, newContent.length - value.length );
+		textarea.focus();
+		onInsert( newContent );
+	} else if ( !! Env.ie && Env.ie >= 11 ) {
```

Comment:
`if ( ! document.queryCommandSupported( 'insertText' ) ) {`

Reply:
This is not "in sync" with the inline comment anymore.
For the post editor toolbar, I only needed to target IE11 to keep the undo history.
Here, though, there was another issue with the cursor positioning that doesn't work as expected on both FF and IE11+.
I believe it didn't happen on the post editor because, with TinyMCE around, there is plenty of stuff happening on content change, that also "accidentally" helped me with the toolbar.

So, in other words, we would not need to check for `insertText`, but for IE11 and Edge instead.